### DISCUSSION TITLE
Has MicroPython 1.20.0 been compiled on ESP32-S3?

### DISCUSSION THREAD

**Discussion by Mars-CN** [micropython/micropython#12379] [ESP32]
Previously used MicroPython 1.19.0+IDF4.4.5, but now there are many compilation issues after switching to IDF5.2+MicroPython 1.20.0.

For example, TinyUSB and ADC have already been completed. But in the end, the following error occurred and there was no way to start:
`
[731/1336] Generating ../../genhdr/qstr.i.last
In file included from /home/mars/esp/micropython-master/ports/esp32/machine_dac.c:37:
/home/mars/esp/esp-idf-master/components/driver/deprecated/driver/dac.h:16:2: warning: #warning "The legacy DAC driver is deprecated, please use `driver/dac_oneshot.h`, `driver/dac_cosine.h` or `driver/dac_continuous.h` instead" [-Wcpp]
   16 | #warning "The legacy DAC driver is deprecated, please use `driver/dac_oneshot.h`, `driver/dac_cosine.h` or `driver/dac_continuous.h` instead"
      |  ^~~~~~~
/home/mars/esp/micropython-master/ports/esp32/machine_uart.c:303:13: warning: 'UART_FIFO_LEN' macro is deprecated, please use 'UART_HW_FIFO_LEN' instead
  303 |     uart_set_hw_flow_ctrl(self->uart_num, self->flowcontrol, UART_FIFO_LEN - UART_FIFO_LEN / 4);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mars/esp/micropython-master/ports/esp32/machine_uart.c:303:13: warning: 'UART_FIFO_LEN' macro is deprecated, please use 'UART_HW_FIFO_LEN' instead
[1084/1336] Generating ../../genhdr/root_pointers.collected
Root pointer registrations not updated
[1090/1336] Generating ../../genhdr/root_pointers.h
FAILED: genhdr/root_pointers.h /home/mars/esp/micropython-master/ports/papermoo/build/genhdr/root_pointers.h 
cd /home/mars/esp/micropython-master/ports/papermoo/build/esp-idf/main_esp32s3 && /home/mars/.espressif/python_env/idf5.2_py3.10_env/bin/python /home/mars/esp/micropython-master/py/make_root_pointers.py /home/mars/esp/micropython-master/ports/papermoo/build/genhdr/root_pointers.collected > /home/mars/esp/micropython-master/ports/papermoo/build/genhdr/root_pointers.h
Traceback (most recent call last):
  File "/home/mars/esp/micropython-master/py/make_root_pointers.py", line 55, in <module>
    main()
  File "/home/mars/esp/micropython-master/py/make_root_pointers.py", line 50, in main
    root_pointers = find_root_pointer_registrations(args.file[0])
  File "/home/mars/esp/micropython-master/py/make_root_pointers.py", line 25, in find_root_pointer_registrations
    with io.open(filename, encoding="utf-8") as c_file_obj:
FileNotFoundError: [Errno 2] No such file or directory: '/home/mars/esp/micropython-master/ports/papermoo/build/genhdr/root_pointers.collected'
[1091/1336] Generating ../../genhdr/qstrdefs.collected.h
QSTR not updated
[1096/1336] Building C object esp-idf/espressif__mdns/CMakeFiles/__idf_espressif__mdns.dir/mdns.c.obj
ninja: build stopped: subcommand failed.
ninja failed with exit code 1, output of the command is in the /home/mars/esp/micropython-master/ports/papermoo/build/log/idf_py_stderr_output_79422 and /home/mars/esp/micropython-master/ports/papermoo/build/log/idf_py_stdout_output_79422
`

**Comment by Mars-CN** [comment]
MICROPY has been added_ PY_ Machine_ DAC set to 0, but the following error still occurred during the ld phase!
`/home/mars/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/bin/../lib/gcc/xtensa-esp32s3-elf/12.2.0/../../../../xtensa-esp32s3-elf/bin/ld: esp-idf/main_esp32s3/libmain_esp32s3.a(machine_dac.c.obj):(.literal.mdac_write+0x8): undefined reference to `dac_output_voltage'
/home/mars/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/bin/../lib/gcc/xtensa-esp32s3-elf/12.2.0/../../../../xtensa-esp32s3-elf/bin/ld: esp-idf/main_esp32s3/libmain_esp32s3.a(machine_dac.c.obj):(.literal.mdac_make_new+0xc): undefined reference to `dac_output_enable'
/home/mars/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/bin/../lib/gcc/xtensa-esp32s3-elf/12.2.0/../../../../xtensa-esp32s3-elf/bin/ld: esp-idf/main_esp32s3/libmain_esp32s3.a(machine_dac.c.obj): in function `mdac_write':
/home/mars/esp/micropython-master/ports/esp32/machine_dac.c:90: undefined reference to `dac_output_voltage'
/home/mars/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/bin/../lib/gcc/xtensa-esp32s3-elf/12.2.0/../../../../xtensa-esp32s3-elf/bin/ld: esp-idf/main_esp32s3/libmain_esp32s3.a(machine_dac.c.obj): in function `mdac_make_new':
/home/mars/esp/micropython-master/ports/esp32/machine_dac.c:62: undefined reference to `dac_output_enable'
/home/mars/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/bin/../lib/gcc/xtensa-esp32s3-elf/12.2.0/../../../../xtensa-esp32s3-elf/bin/ld: /home/mars/esp/micropython-master/ports/esp32/machine_dac.c:72: undefined reference to `dac_output_voltage'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.`

**Comment by andrewleech** [comment]
The nightly builds here are all from master with IDF 5.0.2
https://micropython.org/download/ESP32_GENERIC_S3/

Espressif are deprecating lots of software apis in IDF 5.x so it's not surprising that 5.2 doesn't work yet; pr's for new api support would definitely be welcomed :-)
